pageserver/src/tenant.rs

```diff
@@ -2160,13 +2160,23 @@ impl Tenant {
         let new_timeline = self
             .prepare_timeline(
                 dst_id,
-                metadata,
+                &metadata,
                 timeline_uninit_mark,
                 false,
                 Some(Arc::clone(src_timeline)),
             )?
             .initialize_with_lock(&mut timelines, true, true)?;
         drop(timelines);
+
+        // Root timeline gets its layers during creation and uploads them along with the metadata.
+        // A branch timeline though, when created, can get no writes for some time, hence won't get any layers created.
+        // We still need to upload its metadata to avoid branch loss on (re)attach.
+        if let Some(remote_client) = new_timeline.remote_client.as_ref() {
+            remote_client
+                .schedule_index_upload_for_metadata_update(&metadata)
```

Comment:
Oh, sorry, now I've read it again and you mean that there could be multiple pageservers in action — in this case, I don't think this PR solves this issue indeed, but there has to be a more global approach then, since root timeline creation has the same issue.

Reply:
Do you see an easy way to do that for the branches? I fear that it would take a while (and would definitely need test adjustments, if not the new tests), hence propose to create a standalone issue and fix both alltogether.

Meanwhile, if this gets merged as is, I can proceed with fixing another bug: https://github.com/neondatabase/neon/pull/3843

Reply:
https://github.com/neondatabase/neon/issues/3865